feature/platform/tests/telemetry_inventory_test/telemetry_inventory_test.go

```diff
@@ -581,20 +581,18 @@ func TestControllerCardEmpty(t *testing.T) {
 
 // validateSubcomponentsExistAsComponents checks that if the given component has subcomponents, that
 // those subcomponents exist as components on the device (i.e. the leafref is valid).
-func validateSubcomponentsExistAsComponents(c *oc.Component, components []*oc.Component, t *testing.T) {
-	// Note that this will get the subcomponents regardless of whether they are stored in
-	// subcomponent/name or subcomponent/state/name.
-	for _, s := range c.Subcomponent {
-		sName := s.GetName()
-		if sName != "" {
-			for _, component := range components {
-				if component.GetName() == sName {
-					continue
-				}
-			}
-			t.Errorf("Subcomponent %s not found in list of components", sName)
-		} else {
-			t.Errorf("Component %s has empty subcomponent name", c.GetName())
+func validateSubcomponentsExistAsComponents(c *oc.Component, components []*oc.Component, t *testing.T, dut *ondatra.DUTDevice) {
+	cName := c.GetName()
+	subcomponents := gnmi.LookupAll[*oc.Component_Subcomponent](t, dut, gnmi.OC().Component(cName).SubcomponentAny().State())
```

Comment:
optional:

you can get the map object directly for convenience: gnmi.Lookup(gnmi.OC().Component(cName).SubcomponentMap().State())

You can also do this with components if you like (as long as there is not too much data, in which case you can use `ygnmi.NewBatch`).

Reply:
Thanks for pointing that out! I have updated it to use gnmi.Lookup() with the subcomponentMap().